src/axolotl/monkeypatch/llama_attn_hijack_xformers.py

```diff
@@ -0,0 +1,233 @@
+"""
+Directly copied the code from https://raw.githubusercontent.com/oobabooga/text-generation-webui/main/modules/llama_attn_hijack.py and made some adjustments
```

Comment:
Preferably replace it to john smith's repo for credit which can reference the source

```suggestion
Directly copied the code from https://github.com/johnsmith0031/alpaca_lora_4bit/blob/main/monkeypatch/llama_attn_hijack_xformers.py and made some adjustments
```

Reply:
I actually copied it straight from ooba since it was updated since john smith imported it into his. john smith didn't have sdp_attention.